grails-app/conf/application.yml

```diff
@@ -184,8 +184,10 @@ ala:
 doi:
     service:
         mock: false
-        # ANDS or DATACITE
-        provider: ANDS
+        provider:
+#            # define mapping from provider name to a valid DoiProvider (ANDS or DATACITE)
+            mapping: { ALA: 'DATACITE' }
```

Comment:
Sorry, but prior to review complety: I don't understand why to modify this variable a dictionary. It's not enough like before or `doi.storage.provider` ?

Reply:
The change is to support a mapping between an alternate provider name to one of the `DoiProvider` enum instances. 

The `doi.storage.provider` variable was only used to define defaultValue of the `Doi.provider` domain property.
https://github.com/AtlasOfLivingAustralia/doi-service/blob/fc753cfe4cdad465dcbd356153306a741aa86d87/grails-app/domain/au/org/ala/doi/Doi.groovy#L68
The `defaultValue` is only used when building the database schema. The doi-service uses flyway to manage the DB schema so `defaultValue` is redundant.